docs/01-module-block-1/02-learning-goals.adoc

```diff
@@ -2,39 +2,83 @@
 
 // tag::DE[]
 [[LZ-1-1]]
+==== LZ 1-1: Daten
 
-- Was sollen die Teilnehmer können?
-   * Informationen als unveränderliche Werte repräsentieren
-   * zentrale Funktionalität als Funktionen formulieren
-   * für diese Funktionen Typen konstruieren
+* die Rolle von systematischer Datenanalyse in der Domänenmodellierung
+  verstehen
+* Informationen systematisch als Daten repräsentieren können
+* den Unterschied zwischen zusammengesetzten und gemischten Daten
+  verstehen
+* die Rolle von Selbstbezügen in der Datenrepräsentation verstehen
+* bei der Datenanalyse Selbstbezüge finden und nutzen können
+* den Unterschied zwischen Objekten und Werten verstehen
+* den Unterschied zwischen funktionalen und objektorientierten
+  Datenstrukturen verstehen
 
-- Was sollen die Teilnehmer verstehen?
-  * den Unterschied zwischen dem objektorientierten und dem funktionalen
-Software-Modell
-  * die Rolle von Typen beim funktionalen Software-Design
-  * Module als Alternative zu Klassen
+[[LZ-1-2]]
+==== LZ 1-2: Funktionen
 
-- Was sollen die Teilnehmer kennen?
-  * Beispiele für funktionale System-Architekturen
+* Funktionen systematisch anhand von Anforderungen konstruieren können
+* den Unterschied zwischen reinen Funktionen und zustandsverändernden
+  Prozeduren beziehungsweise objektorientierten Methoden verstehen
+* systematische Abstraktion anwenden können
+* Funktionen durch Abstraktion komponieren können
 
+[[LZ-1-3]]
+==== LZ 1-3: Typen
+
+* die Rolle von Typen bei der Datenanalyse verstehen
+* Typen benutzen können, um Funktionen systematisch zu konstruieren
+* Datenrepräsentationen in sowohl getypten als auch ungetypten
+  Sprachen aus der Datenanalyse herleiten können
+
+[[LZ-1-4]]
+==== LZ 1-4: Modularisierungsmechanismen
+
+* den Unterschied zwischen den Modularisierungsmechanismen in
+  objektorientierter und funktionaler Programmierung verstehen
+* den Modulmechanismus mindestens einer konkreten Sprache
```

Comment:
Das ist mir ein bisschen zu abstrakt. Beispiel: ML-Module und Haskell-Module sind völlig verschiedene Konzepte.

Reply:
Wie praktisch relevant ist diese Abbildung? In der Praxis werden die beiden Konzepte trotz Abbildbarkeit ja doch sehr verschieden gehandhabt.

Man könnte argumentieren, dass ein Schulungsanbieter sich einfach aussuchen kann, welches Modularisierungskonzept vorgestellt wird. Damit wäre ich einverstanden.

Reply:
Does 21d53e6d1540678d0a8cf215bc42fc1a5312cca7 make it better?